internal/intentapi/http_server.go

```diff
@@ -0,0 +1,127 @@
+package intentapi
+
+import (
+	"context"
+	"crypto/tls"
+	"errors"
+	"net/http"
+	"time"
+
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/services/featuremgmt"
+	"github.com/grafana/grafana/pkg/setting"
+)
+
+// HTTPServerConfig is the config for Intent API HTTP server.
+type HTTPServerConfig struct {
+	Enabled         bool
+	ListenAddress   string
+	ServerCert      tls.Certificate
+	ReadTimeout     time.Duration
+	WriteTimeout    time.Duration
+	ShutdownTimeout time.Duration
+}
+
+// NewHTTPServerConfig parses and returns a new HTTPServerConfig from setting.Cfg.
+func NewHTTPServerConfig(cfg *setting.Cfg, featureEnabled bool) (HTTPServerConfig, error) {
+	if !featureEnabled {
+		return HTTPServerConfig{
+			Enabled: false,
+		}, nil
+	}
+
+	sec := cfg.Raw.Section("intentapi.server")
+
+	serverCert, err := tls.LoadX509KeyPair(
+		sec.Key("cert_file_path").MustString(""),
+		sec.Key("key_file_path").MustString(""),
+	)
+	if err != nil {
+		return HTTPServerConfig{}, err
+	}
+
+	c := HTTPServerConfig{}
+	c.Enabled = featureEnabled
+	c.ListenAddress = sec.Key("listen_address").MustString("127.0.0.1:8443")
+	c.ReadTimeout = sec.Key("read_timeout").MustDuration(1 * time.Minute)
+	c.WriteTimeout = sec.Key("write_timeout").MustDuration(1 * time.Minute)
+	c.ShutdownTimeout = sec.Key("shutdown_timeout").MustDuration(1 * time.Minute)
+	c.ServerCert = serverCert
+
+	return c, nil
+}
+
+// Handler handles Intent API requests.
+type Handler interface {
+	http.Handler
+}
+
+// HTTPServer is an HTTP server that exposes Intent API.
+type HTTPServer struct {
+	config  HTTPServerConfig
+	handler Handler
+	logger  log.Logger
+}
+
+// ProvideHTTPServer provides a new HTTP server for Intent API.
+func ProvideHTTPServer(cfg *setting.Cfg, features featuremgmt.FeatureToggles, handler Handler) (*HTTPServer, error) {
+	en := features.IsEnabled("intentapi")
```

Comment:
I think that we have [to register the flag in featuremgmt](https://github.com/grafana/grafana/pull/44541#discussion_r793783080) too